doc/userguide/lua/libs/rule.rst

```diff
@@ -0,0 +1,70 @@
+Rule
+====
+
+Rule details for an alert are exposed to Lua scripts with the
+``suricata.rule`` library, for example::
+
+  local rule = require("suricata.rule")
+
+Rule Setup
+----------
+
+For use in Suricata Lua rules, no additional setup is required.
+
+Output Setup
+------------
+
+For use in Suricata Lua output scripts, some additional setup is
+required::
+
+  function init(args)
+      return {
+          type = "packet",
+          filter = "alerts",
+      }
+  end
+
+Getting a Rule Instance
+-----------------------
+
+To obtain a rule object, use the ``get_rule()`` function on the
+``rule`` library::
+
+  local sig = rule.get_rule()
+
+Rule Methods
+------------
+
+``sid()``
+^^^^^^^^^
+
+Returns the signature ID of the rule.
+
+``gid()``
+^^^^^^^^^
+
+Returns the generator ID of the rule.
+
+``rev()``
+^^^^^^^^^
+
+Returns the revision of the rule.
+
+``action()``
+^^^^^^^^^^^^
+
+Returns the action of the rule, for example: `alert`, `pass`.
+
+``msg()``
+^^^^^^^^^
+
+Returns the rule message (``msg``).
+
+``class()``
+^^^^^^^^^^^
+
+Returns the classification name, as well as priority for the rule. For
+example::
+
+  local r = rule.get_rule()
+  local class, prio = r:class()
```

Comment:
I think `.class_name()` and `.class_priority()` might be better?

Reply:
Is this a draft PR as you are not sure of yourself ?

Reply:
What is the priority for a rule ? Is it not the keyword priority ? `https://docs.suricata.io/en/latest/rules/meta.html#priority`

Reply:
> What is the priority for a rule ? Is it not the keyword priority ? `https://docs.suricata.io/en/latest/rules/meta.html#priority`

For most rules, the priority comes from the classification (https://github.com/OISF/suricata/blob/master/etc/classification.config). But this brings up a good point.  Once the rule is loaded, we probably shouldn't be querying the classification for the priority, just the rule directly.  On the change that it has its priority set in the rule.

This will need a changes requested.